Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.37762&attempt_id=718c18cc-544f-42e1-ad35-74787f911393)

Attempted: python2.pkgs.hdbscan, python3.pkgs.hdbscan

<details><summary>Partial log (click to expand)</summary><p>

```
                 from /nix/store/savp4vc8d7i9nmb7wcsdx3prakp37vgv-python2.7-numpy-1.14.2/lib/python2.7/site-packages/numpy/core/include/numpy/arrayobject.h:4,
                 from extension_type_memoryview.cpp:580:
/nix/store/savp4vc8d7i9nmb7wcsdx3prakp37vgv-python2.7-numpy-1.14.2/lib/python2.7/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:15:2: warning: #warning "Using deprecated NumPy API, disable it by " "#defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
 #warning "Using deprecated NumPy API, disable it by " \
  ^~~~~~~
building of '/nix/store/g9ccyb85m5kxbfb3gy26840d90qxsf9k-python2.7-Cython-0.28.1.drv' timed out after 3600 seconds
building of '/nix/store/l60jan1mgs2yd0lnkmsprzjapx4rynv9-python3.6-Cython-0.28.1.drv' timed out after 3600 seconds
cannot build derivation '/nix/store/zn3mx7hvjjslbw2jzz4nw2pr708dzq89-python2.7-hdbscan-0.8.12.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/qf2g8l733swq4bm3yjmzz9aml0b2gl5c-python3.6-hdbscan-0.8.12.drv': 2 dependencies couldn't be built
[31;1merror:[0m build of '/nix/store/qf2g8l733swq4bm3yjmzz9aml0b2gl5c-python3.6-hdbscan-0.8.12.drv', '/nix/store/zn3mx7hvjjslbw2jzz4nw2pr708dzq89-python2.7-hdbscan-0.8.12.drv' failed
```
</p></details>

--- a/pkgs/development/python-modules/hdbscan/default.nix
+++ b/pkgs/development/python-modules/hdbscan/default.nix
@@ -1,0 +1,31 @@
+{ lib
+, buildPythonPackage
+, cython
+, numpy
+, nose
+, scipy
+, scikitlearn
+, pythonOlder
+, fetchPypi
+}:
+
+buildPythonPackage rec {
+  pname = "hdbscan";
+  version = "0.8.12";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0yxi34frg2jwyvjl942qy4gq5pbx8dq4pf4p28d1xah8njchfqir";
+  };
+
+  checkInputs = [ nose ];
+
+  propagatedBuildInputs = [ cython numpy scipy scikitlearn ];
+
+  meta = with lib; {
+    description = "Hierarchical Density-Based Spatial Clustering of Applications with Noise, a clustering algorithm with a scikit-learn compatible API";
+    homepage =  http://github.com/scikit-learn-contrib/hdbscan;
+    license = licenses.bsd3;
+    maintainers = with maintainers; [ ixxie ];
+  };
+}

--- a/pkgs/development/python-modules/hdbscan/default.nix
+++ b/pkgs/development/python-modules/hdbscan/default.nix
@@ -5,7 +5,6 @@
 , nose
 , scipy
 , scikitlearn
-, pythonOlder
 , fetchPypi
 }:
 

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -6342,6 +6342,8 @@ in {
     propagatedBuildInputs = with self; [ requests webob ];
   };
 
+  hdbscan = callPackage ../development/python-modules/hdbscan { };
+
   hmmlearn = callPackage ../development/python-modules/hmmlearn { };
 
   hcs_utils = callPackage ../development/python-modules/hcs_utils { };

